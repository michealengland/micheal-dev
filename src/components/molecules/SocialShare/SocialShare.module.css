.socialShare {
  @apply inline-block;

  & ul {
    @apply flex flex-nowrap flex-row;
  }

  & li {
    @apply bg-primary-purple ml-4 rounded-full overflow-hidden;
  }

  & .icon {
    @apply block bg-primary-purple;

    height: 32px;
    width: 32px;
  }

  & .label {
    @apply sr-only;
  }
}
