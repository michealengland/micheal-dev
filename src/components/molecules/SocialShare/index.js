import React from 'react'
import styles from './SocialShare.module.css'
import Icon from '@components/atoms/Icon'

export default function SocialShare() {
  const socialAccounts = [
    {
      icon: 'twitter',
      label: 'Github',
      url: 'https://github.com/michealengland'
    },
    {
      icon: 'twitter',
      label: 'Linkedin',
      url: 'https://www.linkedin.com/in/michealengland/'
    },
    {
      icon: 'twitter',
      label: 'Twitter',
      url: 'https://twitter.com/mikelikethebike'
    },
  ]

  return (
    <nav className={styles.socialShare} role="navigation" aria-label="social links">
      <ul>
        { socialAccounts.map((account, index) => {
          const {icon, label, url} = account

          return (
            <li key={`social-${index}`}>
              <a href={url} target="_blank" rel="noreferrer">
                <span className={styles.icon}><Icon icon={icon} /></span>
                <span className={styles.label}>{label}</span>
              </a>
            </li>
          )
        }) }
      </ul>
    </nav>
  )
}