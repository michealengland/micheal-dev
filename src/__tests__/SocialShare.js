import React from 'react'
import renderer from 'react-test-renderer'
import SocialShare from '@components/molecules/SocialShare'
describe('SocialShare', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SocialShare />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
