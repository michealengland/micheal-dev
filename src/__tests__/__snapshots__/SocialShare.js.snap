// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialShare renders correctly 1`] = `
<nav
  aria-label="social links"
  className="socialShare"
  role="navigation"
>
  <ul>
    <li>
      <a
        href="https://github.com/michealengland"
        rel="noreferrer"
        target="_blank"
      >
        <span
          aria-hidden={true}
          className="icon"
        >
          <svg
            className="svg-mock"
            height={200}
            viewBox="0 0 32 32"
            width={200}
          >
            <path
              d="M32 32 32 27.5992141 5.00813008 15.9685658 32 4.27504912 32 0 0 13.956778 0 17.9803536 15.8090702 24.9065278z"
            />
          </svg>
        </span>
        <span
          className="label"
        >
          Github
        </span>
      </a>
    </li>
    <li>
      <a
        href="https://www.linkedin.com/in/michealengland/"
        rel="noreferrer"
        target="_blank"
      >
        <span
          aria-hidden={true}
          className="icon"
        >
          <svg
            className="svg-mock"
            height={200}
            viewBox="0 0 32 32"
            width={200}
          >
            <path
              d="M32 32 32 27.5992141 5.00813008 15.9685658 32 4.27504912 32 0 0 13.956778 0 17.9803536 15.8090702 24.9065278z"
            />
          </svg>
        </span>
        <span
          className="label"
        >
          Linkedin
        </span>
      </a>
    </li>
    <li>
      <a
        href="https://twitter.com/mikelikethebike"
        rel="noreferrer"
        target="_blank"
      >
        <span
          aria-hidden={true}
          className="icon"
        >
          <svg
            className="svg-mock"
            height={200}
            viewBox="0 0 32 32"
            width={200}
          >
            <path
              d="M32 32 32 27.5992141 5.00813008 15.9685658 32 4.27504912 32 0 0 13.956778 0 17.9803536 15.8090702 24.9065278z"
            />
          </svg>
        </span>
        <span
          className="label"
        >
          Twitter
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
